contracts/land/LANDRegistry.sol

```diff
@@ -200,18 +223,76 @@ contract LANDRegistry is Storage,
     emit UpdateOperator(assetId, operator);
   }
 
+  //
+  // Estate generation
+  //
+
+  event EstateRegistrySet(address indexed registry);
+
+  function setEstateRegistry(address registry) onlyProxyOwner external {
+    estateRegistry = IEstateRegistry(registry);
+    emit EstateRegistrySet(registry);
+  }
+
+  function createEstate(
+    int[] x,
+    int[] y,
+    address beneficiary
+  )
+    external
+    returns (uint256)
+  {
+    require(x.length == y.length);
+    require(address(estateRegistry) != 0);
+
+    uint256 estateTokenId = estateRegistry.mintNext(beneficiary);
+
+    for (uint i = 0; i < x.length; i++) {
+      uint256 tokenId = _encodeTokenId(x[i], y[i]);
+      _moveToken(
+        _ownerOf(tokenId),
+        address(estateRegistry),
+        tokenId,
+        toBytes(estateTokenId),
+        this,
+        true
+      );
+    }
+
```

Comment:
Could we emit an `CreateEstate` event here?

Reply:
It's a good idea! I didn't add it here because `_mint` will emit a `Transfer`, `emit Transfer(address(0), _to, _tokenId)`, but it'll be harder to identify from the UI.

Reply:
I think that if we use Transfer() we maintain the same behavior as the LANDRegistry. If we are watching the Transfer() event of the EstateRegistry contract we can identify a Transfer from 0x0 as a mint(), right? so no new event needed.